src/Laika.php

```diff
@@ -48,85 +48,84 @@ class Laika
-     * @param string $password        password for the basic authentication.
+     * @param string $environmentName Environment in which the code is being executed.
+     * @param string $url URL for the API server.
+     * @param string $username Username for the basic authentication.
+     * @param string $password Password for the basic authentication.
+     * @param array $features Optional array with features indexed by name. If the array is not
+     *                        provided the features will be fetched from the client.
      */
-    public function __construct($environmentName, $url, $username, $password)
+    public function __construct($environmentName, $url, $username, $password, $features = null)
     {
         $this->client          = new Client();
         $this->environmentName = $environmentName;
         $this->url             = $url;
         $this->username        = $username;
         $this->password        = $password;
+        $this->features        = $features;
     }
 
     /**
-     * Setter for the feature array.
+     * Get all the features.
      *
-     * @param array $features array with features. Each feature is also an array.
+     * @return array $features array with features indexed by name.
      */
-    public function setFeatures($features)
+    public function getFeatures()
     {
-        $this->features = $features;
+        if (!$this->features) $this->preloadFeatures();
+        return $this->features;
     }
 
     /**
      * Retrieves all the existing features through an HTTP request and adds them to the features array.
      *
-     * @return boolean true if function executed as expected, false if problems occurred.
+     * @return boolean True if function executed as expected, false if problems occurred.
      */
-    public function fetchAllFeatures()
+    protected function preloadFeatures()
     {
-        $requestResult = $this->httpRequest('api/features');
-        if ($requestResult === false) {
-            return false;
-        }
+        $features = $this->get('api/features');
 
-        foreach ($requestResult as $featureValue) {
-            $this->features[$featureValue['name']] = $featureValue;
+        $this->features = array();
+        foreach ($features as $feature) {
+            $this->features[$feature['name']] = $feature;
         }
-
-        return true;
     }
 
     /**
-     * Executes HTTP requests.
+     * Performs an HTTP Get requests, returning the body parsed as JSON.
      *
      * @param  string API endpoint.
-     * @return array|boolean returns an array with the information decoded from the json. If problems occurred, returns false.
+     * @return array|boolean Returns an array with the information decoded from the json. If problems
+     *                       occurred, returns false.
      */
-    protected function httpRequest($endpoint)
+    protected function get($endpoint)
```

Comment:
The library we use will raise an exception if the status is not 2XX. Then the client user can decide what to do in that situation. I guess we could add some option to return false by default if there is an error, but this will do for us for now.